sdk/zksync-rs/src/ethereum/mod.rs

```diff
@@ -418,17 +424,20 @@ impl<S: EthereumSigner> EthereumProvider<S> {
                 tx_data,
                 Options::with(|f| {
                     f.gas = Some(U256::from(300000));
+                    f.transaction_type = Some(EIP_1559_TX_TYPE.into());
                     f.value = Some(value);
                     f.gas_price = Some(gas_price)
                 }),
+                None,
```

Comment:
maybe consider having a separate method (sign_prepared_tx_with_blobs?)

this way you'll not have to add these None,None in all these places?

Reply:
it would be a bit less typesafe but a bit more pretty i suppose. i don't have a strong opinion on it

Reply:
I think this problem would also be solved if we would introduce a custom options type.